routing/base/astar_algorithm.hpp

```diff
@@ -13,18 +13,43 @@
 namespace routing
 {
 
-template <typename TVertexType>
+template <typename TWeightType>
+constexpr TWeightType GetWeightZero();
+template <>
+constexpr double GetWeightZero<double>()
+{
+  return 0.0;
+}
+
+// Precision of comparison weights.
+template <typename TWeightType>
+constexpr TWeightType GetWeightEpsilon();
+template <>
+constexpr double GetWeightEpsilon<double>()
+{
+  return 1e-6;
+}
+
+template <typename TWeightType>
+constexpr TWeightType GetWeightMax();
+template <>
+constexpr double GetWeightMax<double>()
```

Comment:
Why not to create a struct, say,
```
struct DoubleWeight
{
   double constexpr kEpsilon = ...;
   double constexpr kMax = ...;
   ...
};
```
Like iterator traits? So, I suggest to hide all these functions in a struct.